packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -40,7 +42,10 @@ export const NetworkSelectionContainer = ({
 
   return (
     <Popover className="relative z-50 w-full lg:w-max">
-      <Popover.Button className="arb-hover flex w-full justify-start rounded-full px-6 py-3 lg:w-max lg:p-0">
+      <Popover.Button
+        disabled={!isEOA}
```

Comment:
Maybe this will be a bit more easy to understand if it was
```suggestion
        disabled={isSmartContractWallet}
```

Because that's sort of the edge case, EOA is the common case

Reply:
In both cases you've mentioned we want to cover the `undefined` case. The use of `isSmartContractWallet` would make the button enabled when `isSmartContractWallet` is `undefined`.

In that case, a user would be able to click the popover button when the bridge is still lazy loading and the account type is not determined yet.

`!isEOA` fixes that because we require the account type check to finish and for it to be EOA, for users to be able to interact with the button.

Reply:
maybe add a comment to explain this?

Reply:
added a comment

Reply:
Can we just be more explicit and say:

```suggestion
        disabled={isSmartContractWallet || typeof isSmartContractWallet === 'undefined'}
```